forcedotcom/source-deploy-retrieve
# Sh/fix deploy sharing rules

### What does this PR do?
SharingRules were not being converted properly by child type due to an incorrect definition in the registry.  This would result in errors thrown when trying to deploy SharingRules using a command like:
`sf project deploy start -m SharingCriteriaRule:Account.Test_Rule1`
Adds conversion snapshot tests to ensure SharingRules can be converted by the SharingRules type or a child type, both to mdapi format and source format.

### What issues does this PR fix or reference?
@W-16554846@

Test Suggestions:
1. convert SharingRules to mdapi format by path.  Ensure the correct directory name (sharingRules) and file contents.
2. convert SharingRules to mdapi format by SharingRules type
3. convert SharingRules to mdapi format by SharingCriteriaRule type
4. deploy SharingRules by SharingRules type
5. deploy SharingRules by SharingCriteriaRule type
6. retrieve SharingRules by SharingRules type
5. retrieve SharingRules by SharingCriteriaRule type
